compiler/src/dotty/tools/dotc/transform/IsInstanceOfChecker.scala

```diff
@@ -116,7 +130,9 @@ object Checkable {
           case defn.ArrayOf(tpE)   => recur(tpE, tpT)
           case _                   => recur(defn.AnyType, tpT)
         }
-      case tpe: AppliedType     => isClassDetermined(X, tpe)(ctx.fresh.setNewTyperState())
+      case tpe: AppliedType     =>
+        isClassDetermined(X, tpe)(ctx.fresh.setNewTyperState()) ||
```

Comment:
Here, you first try without stripping type parameters for performance reasons? Maybe add a comment